### PR TITLE
build: use tsc output code in both cjs and esm formats 

### PR DESCRIPTION
**Pull request by Lxxyx**
Fixes: #126

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "include": ["src"],
   "exclude": ["**/*.spec.ts", "**/*.test.ts"],
   "compilerOptions": {
-    "module": "CommonJS",
+    "module": "ESNext",
```
Comment (Skn0tt) [inline tsconfig.json:5]: Which effects does this have? Can this change be breaking?

**Review by Skn0tt — CHANGES_REQUESTED** [review]
Thanks @Lxxyx for this PR! After looking at the build output of TSDX, I have two questions:

1. is it possible to not bundle the output together? Bundling should be the responsibility of final application build, and having an unbundled output makes debugging easier for us.
2. It currently outputs declaration files for each of our source files, allowing people to import `superjson/accessDeep` or smth. But since the javascript code for that is all in the bundles, that import will break. Is there a way to omit these per-file declaration maps / have them bundled together?

I tried to look them up in the TSDX docs, but couldn't find anything. Maybe you have an idea :D

File: tsconfig.json (hunk 1 above)
Comment (Lxxyx) [reply tsconfig.json:5]: This is required by `tsdx`. I actually just revert the commit of #97

**Comment by Lxxyx** [comment]
@Skn0tt We can do this. Use tsc to output code in both cjs and esm formats, and be forward compatible


```json
{
  "scripts": {
    "build": "npm run build:cjs && npm run build:esm",
    "build:cjs": "tsc",
    "build:esm": "tsc --module es2015 --outDir dist/esm"
  },
  "module": "dist/esm/index.js"
}
```

**Comment by Skn0tt** [comment]
> We can do this. Use tsc to output code in both cjs and esm formats, and be forward compatible

That'd be great!

**Review by Skn0tt — CHANGES_REQUESTED** [review]
> We can do this. Use tsc to output code in both cjs and esm formats, and be forward compatible

That'd be great!

**Review by Skn0tt — APPROVED** [review]
LGTM! Thanks a lot for your work on this ❤️

**Comment by Skn0tt** [comment]
Published in v1.7.3

**Comment by Skn0tt** [comment]
@allcontributors add Lxxyx for code

**Pull request merged**